src/main/java/edu/kpi/testcourse/logic/repositories/UrlRepository.java

```
@@ -0,0 +1,50 @@
+package edu.kpi.testcourse.logic.repositories;
+
+import edu.kpi.testcourse.logic.entities.UrlEntity;
+import java.util.List;
+
+/**
+ * The Data Access Object for the UrlEntity.
+ */
+public interface UrlRepository {
+
+  /**
+   * Saves the UrlEntity in some storage.
+   *
+   * @param email of the user that creates UrlEntity
+   *
+   * @param urlEntity is the entity to be saved
+   *
+   * @return true if the save operation was successful and no entity exists with the same alias
+   */
+  boolean save(String email, UrlEntity urlEntity);
+
+  /**
+   * Finds the one URL Entity instance by alias.
+   *
+   * @param alias by which the search is made
+   *
+   * @return UrlEntity instance
```

Maybe we should document null behavior explicitly or throw some custom exception in case url was not found

---

I will document null behaviour.